omnigibson/action_primitives/starter_semantic_action_primitives.py

```diff
@@ -0,0 +1,1898 @@
+                # overwrite the goal orientation, since it is in absolute frame.
+                controller_config = self.robot._controller_config["arm_" + self.arm]
+                # assert controller_config["motor_type"] == "velocity", "Controller must be in velocity mode"
+                assert controller_config["mode"] == "pose_absolute_ori", "Controller must be in pose_delta_ori mode"
+                # current_pose = self._get_pose_in_robot_frame((self.robot.get_eef_position(), self.robot.get_eef_orientation()))
+                current_quat = self.robot.get_relative_eef_orientation()
+                current_ori = T.quat2axisangle(current_quat)
+                control_idx = self.robot.controller_action_idx["arm_" + self.arm]
+                action[control_idx[3:]] = current_ori
+                if arm_pose_to_keep is not None:
+                    action[control_idx[:3]] = arm_pose_to_keep - self.robot.get_eef_position()
+
+        return action
+
+    def _reset_hand(self):
+        """
+        Yields action to move the hand to the position optimal for executing subsequent action primitives
+
+        Returns:
+            np.array or None: Action array for one step for the robot to reset its hand or None if it is done resetting
+        """
+        if TORSO_FIXED:
+            control_idx = self.robot.arm_control_idx[self.arm]
+        else:
+            control_idx = np.concatenate([self.robot.trunk_control_idx, self.robot.arm_control_idx[self.arm]])
+
+        controller_config = self.robot._controller_config["arm_" + self.arm]
+        if controller_config["name"] == "InverseKinematicsController":
+            indented_print("Resetting hand")
+            reset_eef_pose = self._get_reset_eef_pose()
+            try:
+                yield from self._move_hand_ik(reset_eef_pose)
+            except ActionPrimitiveError:
+                indented_print("Could not do a planned reset of the hand - probably obj_in_hand collides with body")
+                yield from self._move_hand_direct_ik(reset_eef_pose, ignore_failure=True, in_world_frame=False)
+        else:
+            indented_print("Resetting hand")
+            reset_pose = self._get_reset_joint_pos()[control_idx]
+            try:
+                yield from self._move_hand_joint(reset_pose, control_idx)
+            except ActionPrimitiveError:
+                indented_print("Could not do a planned reset of the hand - probably obj_in_hand collides with body")
+                yield from self._move_hand_direct_joint(reset_pose, control_idx, ignore_failure=True)
+    
+    def _get_reset_eef_pose(self):
+        if self.robot_model == "Tiago":
+            return np.array([0.28493954, 0.37450749, 1.1512334]), np.array([-0.21533823,  0.05361032, -0.08631776,  0.97123871])
+        else:
+            raise NotImplementedError
+
+    def _get_reset_joint_pos(self):
+        reset_pose_fetch = np.array(
```

Comment:
is this different than the hardcoded reset pose for Tiago and Fetch in their respective classes?

Reply:
Yes, unfortunately. For now we'll keep this here since it's very use-specific.